.github/workflows/unit-tests.yaml

```
@@ -0,0 +1,33 @@
+name: CI/CD Action
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          tags: desktop-business-app
+          # We could give some condition later to push the image to the registry
+          push: false
+          load: true
+
+      - name: Run Program
+        run: |
+            docker run desktop-business-app
```

```suggestion
    runs-on: ubuntu-latest
    container: NAME/IMAGE:VERSION

    steps:
      - uses: actions/checkout@v3
      - name: Build Program
        run: |
          cmake -B ${{ env.BUILD_DIR }} -DCMAKE_BUILD_TYPE=Release .
          cmake --build ${{ env.BUILD_DIR }} --parallel 4
      - name: Run Program
        run: ./desktop-business-app
```

If your dependencies will not change often it's better to use a build-in `container` param. 
Before using it you need to push your docker image to docker registry (docker hub).